src/java/main/org/apache/zookeeper/server/auth/KerberosName.java

```diff
@@ -356,6 +358,9 @@ String apply(String[] params) throws IOException {
    */
   public static void setConfiguration() throws IOException {
```

Comment:
thrown IOException should be logged. So can you include this exception where it is re thrown i.e. 
 `
...
...
setConfiguration();
    }
    catch (IOException e) {
      throw new IllegalArgumentException("Could not configure Kerberos principal name mapping.", e);
    }`

Reply:
I was thinking this could be a problem when rules are separated with new line character but seems we can configure rules without new line character like security.auth_to_local= RULE:[2:$1] DEFAULT in zoo.cfg. Based this and already discussed facts, changes proposed in PR are not required. We can close this PR